tests/functional/WSP/test_wsp.py

```diff
@@ -0,0 +1,135 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of hepcrawl.
+# Copyright (C) 2017 CERN.
+#
+# hepcrawl is a free software; you can redistribute it and/or modify it
+# under the terms of the Revised BSD License; see LICENSE file for
+# more details.
+
+"""Functional tests for WSP spider"""
+
+from __future__ import absolute_import, print_function, unicode_literals
+
+import pytest
+import json
+import os
+import shutil
+
+from scrapyd_api import ScrapydAPI
+from time import sleep
+
+from hepcrawl.testlib.tasks import app
+from hepcrawl.testlib.celery_monitor import CeleryMonitor
+
+
+def get_crawler_instance(crawler_host, *args, **kwargs):
+    """Return current crawler instance."""
+    return ScrapydAPI(
+        crawler_host,
+        *args,
+        **kwargs
+    )
+
+
+def override_generated_fields(record):
+    record['acquisition_source']['datetime'] = u'2017-04-03T10:26:40.365216'
+    record['acquisition_source']['submission_number'] = u'5652c7f6190f11e79e8000224dabeaad'
+
+    return record
+
+
+@pytest.fixture(scope="module")
+def expected_results():
+    file_name = 'fixtures/wsp_smoke_records.json'
+    responses_dir = os.path.dirname(os.path.realpath(__file__))
+    response_file = os.path.join(responses_dir, file_name)
+
+    with open(response_file) as fd:
+        expected_data = json.load(fd)
+
+    return expected_data
+
+
+@pytest.fixture(scope="function")
+def set_up_ftp_environment():
+    netrc_location = os.path.join(os.path.dirname(
+        os.path.realpath(__file__)),
+        'fixtures/ftp_server/.netrc'
+    )
+
+    yield {
+        'CRAWLER_HOST_URL': 'http://scrapyd:6800',
+        'CRAWLER_PROJECT': 'hepcrawl',
+        'CRAWLER_ARGUMENTS': {
+            'ftp_host': 'ftp_server',
+            'ftp_netrc': netrc_location,
+        }
+    }
+
+    cleaner()
+
+
+@pytest.fixture(scope="function")
+def set_up_local_environment():
+    package_location = os.path.join(os.path.dirname(
+        os.path.realpath(__file__)),
+        'fixtures/ftp_server/WSP/'
+    )
+
+    yield {
+        'CRAWLER_HOST_URL': 'http://scrapyd:6800',
+        'CRAWLER_PROJECT': 'hepcrawl',
+        'CRAWLER_ARGUMENTS': {
+            'package_path': package_location,
+        }
+    }
+
+    cleaner()
+    cleaner(package_location + 'IDAQPv20i01-03160015-1510863')
+
+
+def cleaner(path='/tmp/WSP/'):
```

Comment:
This is a bad name for a function, cleaner gives the impression to be an object, that you call on (it's an entity that has meaning by itself) while this is actually a function, maybe rename to something like `clean_dir`?